src/full/Agda/TypeChecking/Errors.hs

```diff
@@ -191,6 +191,17 @@ prettyDisambProj = prettyDisamb $ lastMaybe . filter (noRange /=) . map nameBind
 prettyDisambCons :: MonadPretty m => QName -> m Doc
 prettyDisambCons = prettyDisamb $ Just . nameBindingSite . qnameName
 
+-- | If we try to apply a function whose type is not a function type
+--   but also not blocked on anything, then the user might by accident
+--   have supplied too many arguments.
+--
+tooManyArgumentsHintIfReallyNotBlocked :: MonadPretty m => I.Type -> m Doc
+tooManyArgumentsHintIfReallyNotBlocked t = do
+  reduceB t >>= \case
```

Comment:
Surely we have this blocking information where we throw the error, instead of having to reduce the type while pretty-printing it?

Reply:
Yes one could optimize this.

However, I lack faith in this hint, as instead of _too many arguments_, the "real" reason could be that a hidden argument was used visibly.  So, before throwing this hint at the user, we should at least also check whether we inserted some trailing hidden arguments.